examples/grpc-server-with-prometheus/client/client.go

```diff
@@ -0,0 +1,39 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	pb "github.com/grpc-ecosystem/go-grpc-prometheus/examples/grpc-server-with-prometheus/protobuf"
+	"github.com/grpc-ecosystem/go-grpc-prometheus/examples/grpc-server-with-prometheus/util"
+)
+
+func main() {
+	// Create a gRPC channel to communicate with the server
+	conn, err := grpc.Dial(
+		"localhost:"+strconv.FormatInt(util.SERVER_PORT, 10),
+		grpc.WithInsecure(),
+		// For grpc_prometheus
+		grpc.WithUnaryInterceptor(grpc_prometheus.UnaryClientInterceptor),
+		grpc.WithStreamInterceptor(grpc_prometheus.StreamClientInterceptor))
+	defer conn.Close()
```

Comment:
In case of error, `conn` can be potentially (and is allowed to be) just `nil`, so it will trigger panic here.

Let's maybe defer after checking error,  the connection is definitely not open in case of error.

Reply:
I strongly agree this suggestion.